File: services/aas-designer/AasDesignerApi/Model/Client/OrganisationUebersichtBenutzerDto.cs
namespace AasDesignerApi.Model.Client
{
    public class OrganisationUebersichtBenutzerDto
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Vorname { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Telefon { get; set; } = string.Empty;
        public bool Geloescht { get; set; }
        public bool IsSystemUser { get; set; }
        public bool AccountAktiv { get; set; }
        public bool CreatedBySso { get; set; }
        public List<string> BenutzerRollen { get; set; } = new List<string>();
    }
}

File: frontend/libs/aas-designer-core/src/lib/organisation/model/organisation-uebersicht-benutzer-dto.ts
export class OrganisationUebersichtBenutzerDto {
  id: number = 0;
  name: string = '';
  vorname: string = '';
  _email: string = '';
  telefon: string = '';
  geloescht: boolean = false;
  accountAktiv: boolean = false;
  benutzerRollen: string[] = [];
  isSystemUser: boolean = false;

  static fromDto(dto: any) {
    const benutzer = new OrganisationUebersichtBenutzerDto();
    Object.assign(benutzer, dto);

    return benutzer;
  }

  toDto() {
    return {
      id: this.id,
      name: this.name,
      vorname: this.vorname,
      email: this.email,
      telefon: this.telefon,
      benutzerRollen: this.benutzerRollen,
      accountAktiv: this.accountAktiv,
    };
  }

  set email(value: string | number) {
    this._email = value.toString();
  }

  get email(): string {
    return this._email.toString();
  }
}
